conans/test/unittests/util/tools_test.py

```diff
@@ -661,9 +662,10 @@ def error_url():
 
         out = TestBufferConanOutput()
         # Test: File name cannot be deduced from '?file=1'
-        with six.assertRaisesRegex(self, ConanException,
-                                   "Cannot deduce file name from url. Use 'filename' parameter."):
+        with self.assertRaises(ConanException) as error:
             tools.get("http://localhost:%s/?file=1" % thread.port, output=out)
+        self.assertIn("Cannot deduce file name from the url: 'http://localhost:{}/?file=1'."
```

Comment:
better access error.exception in the context

Reply:
I can't. Once raised, the context will be exited, that's why I need to access out of the context.